kategory-docs/docs/docs/patterns/glossary/README.md

```diff
@@ -0,0 +1,145 @@
+This list isn't parametrized in any generic value, so it cannot be considered a type constructor anymore.
+
+Type constructors are useful when matched with typeclasses because they help us represent non-parametrized values.
+As type constructors is not a first class feature in Kotlin, we use an interface `Hk<F, A>` to represent them.
+Hk stands for Higher Kind, which is the name of the feature that allows working with type constuctors as first class.
+
+#### Higher Kinds
+
+In a Higher Kind with the shape `HK<F, A>`, if `A` is the type of the content then `F` has to be the type of the container.
+A malformed container would use the whole type constructor, duplicating the type ~~HK<Option<A>, A>~~.
+What KΛTEGORY does instead is define a surrogate type, known as the witness, that's not parametrized to represent `F`.
+These types are named same as the container, suffixed by HK, as in `OptionHK` or `ListKWHK`.
+
+You can read more about Higher Kinds and type constructors in [KindedJ's README](https://github.com/KindedJ/KindedJ#rationale)
+
+#### Using Higher Kinds with typeclasses
+
+When coupled with typeclasses, we can now define mapability using ([`Functor`]({{ '/docs/typeclasses/functor' | relative_url }})) for any `ListKW`.
+
+```kotlin
+interface Functor<F>: Typeclass {
+  fun <A, B> map(fa: Hk<F, A>, f: (A) -> B): HK<F, B>
+}
+```
+
+```kotlin
+object ListKWFunctorInstance : Functor<ListKWHK> {
+  override fun <A, B> map(fa: Hk<ListKWHK, A>, f: (A) -> B): ListKW<B> {
+    val list: ListKW<A> = fa.ev()
+    return list.map(f)
+  }
+}
+```
+
+You can see a function `ev()` used to access the `map()` function that already exists in `ListKW`.
+This is because we need to safely downcast from `Hk<ListKWHK, A>` to `ListKW`, and `ev()` is a global function defined to do so.
+
+The function `ev()` is already defined for all datatypes in KΛTEGORY. If you're creating your own datatype that's also a type constructor and would like to create all these helper types and functions,
+you can do so simply by annotating it as `@higerkind`, and using KΛTEGORY's [annotation processor](https://github.com/kategory/kategory#additional-setup) will create them for you.
+
+#### Using Higher Kinds and typeclasses with functions
+
+Higher kinds can also be used to represent functions that are parametrized on type constructors.
+As long as you have a typeclass that can provide you with the behavior required to use such datatypes, you're good to go!
+
+Let's use the typeclass ([`Applicative`]({{ '/docs/typeclasses/applicative' | relative_url }})), that contains the constructor function `pure()`.
+
+```kotlin
+interface Applicative<F>: Functor<F>, Typeclass {
+  fun <A> pure(a: A): HK<F, A>
+}
+
+object ListKWApplicativeInstance : ListKWFunctorInstance, Applicative<ListKWHK> {
+  override fun <A> pure(a: A): HK<F, A> = listOF(a)
+}
+
+inline fun <reified F> randomUserStructure(f: (Int) -> User, AP: Applicative<F> = applicative<F>()) =
```

Comment:
Important to note why we need `reified` or will this be added later with @JorgeCastilloPrz suggestion to clear up the HK paraphernalia?

Reply:
Well that's already described [here](https://github.com/kategory/kategory/blob/ebbdeee8393a2cb30bbc323a3560dcdfe6ef6a2a/kategory-docs/docs/docs/patterns/glossary/README.md#type-constructors). By looking at the PR I don't have a clear vision about where it's gonna be located, but that's kind of the more accurate explanation about how we emulate HKs that I was asking for. Even though, I think it should be part of the most important docs on the library (the ones on the top of the side menu), since the user really needs to understand this before digging on datatypes or posts.

Reply:
> Well that's already described here ?

There is no explanation as to why `F` needs to be `reified` in order to lookup the instance or that it is a requirement for looking up the instance? I would like to see a note that `F` needs to be `reified` if you want to be able to lookup the instance.

Reply:
I'll add the note now